### PR TITLE
explanation for commander/agent states

### PR DESCRIPTION
**Pull request by AcidSepp**

### CODE REVIEW

File: LaTeX/Parts/Chapter3.tex [hunk 1]
```diff
@@ -1,14 +1,25 @@
 \chapter{Execution}
+To start a Varroa test the User needs to start a Commander and at least one Agent.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
To start a Varroa test the user needs to start a Commander and at least one Agent.
```

File: LaTeX/Parts/Chapter3.tex [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+The Agents periodically try to establish a connection to the Commander.
+When all Agents are connected the execution of the scenario starts.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
Once all Agents are connected the execution of the scenario starts.
```

File: LaTeX/Parts/Chapter3.tex [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
+\section{Commander}
 \begin{figure}[H]
 	\begin{center}
 	\includegraphics[scale=0.9]{Resources/PDF/CommanderStates}
 	\caption{Commander States}
 	\label{pic:CommanderStates}
 	\end{center}
 \end{figure}
+After starting a Varroa instance that is configured as a Commander (see \ref{sec:commanderConfig}) it waits for the agents in the network to connect.
+Until all agents are successfully connected the Commander remains in the 'Waiting for Agents' state.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
Until all agents are successfully connected the Commander remains in the \emph{Waiting for Agents} state.
```
same with the other states.

File: LaTeX/Parts/Chapter3.tex [hunk 4]
```diff
@@ -1,14 +1,25 @@
...
-\subsection{Commander}
+\section{Commander}
 \begin{figure}[H]
 	\begin{center}
 	\includegraphics[scale=0.9]{Resources/PDF/CommanderStates}
 	\caption{Commander States}
 	\label{pic:CommanderStates}
 	\end{center}
 \end{figure}
+After starting a Varroa instance that is configured as a Commander (see \ref{sec:commanderConfig}) it waits for the agents in the network to connect.
```
Comment (SgtSilvio) [inline LaTeX/Parts/Chapter3.tex]: "in the network"?

File: LaTeX/Parts/Chapter3.tex [hunk 5]
```diff
@@ -1,14 +1,25 @@
...
+Until all agents are successfully connected the Commander remains in the 'Waiting for Agents' state.
+As the last missing agent connects the Commander switches its state to 'All Agents Connected'.
+Then it begins to parse the Scenario's XML file and within finishing it starts to distribute the Scenario's first stage among the agents.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
Then it parses the Scenario's XML. As soon is this is done, the Commander starts to distribute the Scenario's first stage to the agents.
```

File: LaTeX/Parts/Chapter3.tex [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
+In doing so the Commander changes its state to 'Scenario in Progress'.
+It remains in this stage until all stages are executed successfully and then transfers its state into 'Scenario Finished'.
+Contrary to this case a faulty execution of a stage results in the failure of the whole Scenario, logically the Commanders state is now 'Scenario Failed'.
+Regardless of success or failure the Commander terminates.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex]: useless information

File: LaTeX/Parts/Chapter3.tex [hunk 7]
```diff
@@ -1,14 +1,25 @@
...
 	\caption{Commander States}
 	\label{pic:CommanderStates}
 	\end{center}
 \end{figure}
+After starting a Varroa instance that is configured as a Commander (see \ref{sec:commanderConfig}) it waits for the agents in the network to connect.
+Until all agents are successfully connected the Commander remains in the 'Waiting for Agents' state.
+As the last missing agent connects the Commander switches its state to 'All Agents Connected'.
+Then it begins to parse the Scenario's XML file and within finishing it starts to distribute the Scenario's first stage among the agents.
+In doing so the Commander changes its state to 'Scenario in Progress'.
+It remains in this stage until all stages are executed successfully and then transfers its state into 'Scenario Finished'.
```
Comment (SgtSilvio) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
It remains in this state until all stages are executed successfully and then transfers its state into 'Scenario Finished'.
```

File: LaTeX/Parts/Chapter3.tex [hunk 8]
```diff
@@ -18,9 +29,16 @@ \subsection{Agent}
 	\label{pic:AgentStates}
 	\end{center}
 \end{figure}
+Upon starting a Varroa instance as an Agent it tries to connect to the configured Commander (see \ref{sec:agentConfig}).
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
Upon starting a Varroa instance as an Agent, it tries to connect to the configured Commander (see \ref{sec:agentConfig}).
```

File: LaTeX/Parts/Chapter3.tex [hunk 9]
```diff
@@ -1,14 +1,25 @@
...
 	\label{pic:CommanderStates}
 	\end{center}
 \end{figure}
+After starting a Varroa instance that is configured as a Commander (see \ref{sec:commanderConfig}) it waits for the agents in the network to connect.
+Until all agents are successfully connected the Commander remains in the 'Waiting for Agents' state.
+As the last missing agent connects the Commander switches its state to 'All Agents Connected'.
+Then it begins to parse the Scenario's XML file and within finishing it starts to distribute the Scenario's first stage among the agents.
+In doing so the Commander changes its state to 'Scenario in Progress'.
+It remains in this stage until all stages are executed successfully and then transfers its state into 'Scenario Finished'.
+Contrary to this case a faulty execution of a stage results in the failure of the whole Scenario, logically the Commanders state is now 'Scenario Failed'.
```
Comment (SgtSilvio) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
Contrary to this case a faulty execution of a stage results in the failure of the whole Scenario, logically the Commanders state is then 'Scenario Failed'.
```

File: LaTeX/Parts/Chapter3.tex [hunk 10]
```diff
@@ -18,9 +29,16 @@ \subsection{Agent}
 	\label{pic:AgentStates}
 	\end{center}
 \end{figure}
+Upon starting a Varroa instance as an Agent it tries to connect to the configured Commander (see \ref{sec:agentConfig}).
+It periodically attempts to establish the connection until it succeeds.
+Then it waits for incoming Chunks from the Commander and executes them, implementing the handshake protocol introduced in \ref{sec:chunkDistribution}.
+It does so until it loses the connection to the commander.
+In contrast to the Commander the Agents life-cycle does not stop after the execution of a Scenario.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex:37]: new paragraph

File: LaTeX/Parts/Chapter3.tex [hunk 11]
```diff
@@ -18,9 +29,16 @@ \subsection{Agent}
 	\label{pic:AgentStates}
 	\end{center}
 \end{figure}
+Upon starting a Varroa instance as an Agent it tries to connect to the configured Commander (see \ref{sec:agentConfig}).
+It periodically attempts to establish the connection until it succeeds.
+Then it waits for incoming Chunks from the Commander and executes them, implementing the handshake protocol introduced in \ref{sec:chunkDistribution}.
+It does so until it loses the connection to the commander.
```
Comment (sauroter) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
The Agent continues doing so, until it loses the connection to the commander.
```

**Review by sauroter — CHANGES_REQUESTED** [review]
Comments are directly in the files.

File: LaTeX/Parts/Chapter3.tex (hunk 11 above)
Comment (SgtSilvio) [inline LaTeX/Parts/Chapter3.tex]:
```suggestion
It does so until the Commander closes the connection.
```

Comment (RobinAtherton) [reply LaTeX/Parts/Chapter3.tex]: Used Georg's Request for this line.

**Review by sauroter — APPROVED** [review]

**Pull request merged**